db/system_keyspace.cc

```diff
@@ -1585,12 +1585,18 @@ future<system_keyspace::local_info> system_keyspace::load_local_info() {
         if (row.has("cluster_name")) {
             ret.cluster_name = row.get_as<sstring>("cluster_name");
         }
+        if (row.has("data_center")) {
+            ret.dc = row.get_as<sstring>("data_center");
+        }
+        if (row.has("rack")) {
```

Comment:
Having a rack must also require having a dc, I'd nest this block in the block above.
Not sure if the condition should for both dc and rack though, or if the code is prepared for only the dc present, with no rack.